pkg/controllers/allocation/batch.go

```diff
@@ -0,0 +1,155 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package allocation
+
+import (
+	"sync"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+)
+
+const (
+	BatchCacheTTL             = 1 * time.Hour
+	BatchCacheCleanupInterval = 2 * time.Hour
+)
+
+type Batcher struct {
```

Comment:
Thoughts on testing this?

Reply:
I have some basic unit tests for this that I can check in. We'll have some confidence that they work since this will be utilized in the suite tests.  Did you have any specific automated tests in mind?

Reply:
This won't be utilized by suite tests. We call reconcile directly now. Thinking on this, the batching window will massively slow our tests down :P

Reply:
unless we configure them to be microsecond Timeouts. That's what I did on my little unit tests

Reply:
I think you can simply fake time in golang. Happy to discuss offline.